painttiming.bs

```diff
@@ -62,9 +89,42 @@ Formally, we consider the user agent to have "rendered" a document when it has p
 
     NOTE: The rendering pipeline is very complex, and the timestamp should be the latest timestamp the user agent is able to note in this pipeline (best effort). Typically the time at which the frame is submitted to the OS for display is recommended for this API.
 
+A [=generated content pseudo-element=] is a <dfn>paintable pseudo-element</dfn> when all of the following apply:
+* The pseudo-element's [=used=] [=visibility=] is <code>visible</code>.
+* The pseudo-element's [=used=] [=opacity=] is greater than zero.
+* The pseudo-element generates a non-empty [=box=].
+
+An [=element=] is <dfn export>contentful</dfn> when one or more of the following apply:
+* The element has a [=text node=] child, with at least one character, excluding [=document white space characters=].
+* The element is a [=replaced element=] representing a [=loaded=] image.
+* The element has a [=loaded=] [=background-image=], and its [=used=] [=background-size=] has non-zero width and height values.
+
+    NOTE: a gradient is not considered a loaded background image, and thus would not make an element count as contentful.
+
+* The element is a [=canvas=] with its [=context mode=] set to any value other than <code>none</code>.
+* The element is an [=svg element with rendered descendants=].
+* The element is an [=originating element=] for a [=paintable pseudo-element=] that represents a [=loaded=] image or text with at least one character excluding [=document white space characters=].
+
+<dfn>Paintable Bounding Rect</dfn>: The paintable bounding rect is the [=element=]'s bounding rect, which would be the result of running the [=getBoundingClientRect=] algorithm on the [=element=], clipped by ancestors with <code>overflow: clip</code>.
```

Comment:
A better way to frame this is to define Paintable Bounding Rect as an algorithm that takes in an [=element=] |el| and returns a value, and then call this algorithm with an element as input.

The current definition would require monkey patching element with a paintable bounding rect property, which is messy.